src/deriveAddress.c

```diff
@@ -207,98 +314,26 @@ static void deriveAddress_display_ui_runStep()
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
 	UI_STEP(DISPLAY_UI_STEP_WARNING) {
-		#ifdef HAVE_BAGL
-		ui_displayPaginatedText(
-		        "Unusual request",
-		        "Proceed with care",
-		        this_fn
-		);
-		#elif defined(HAVE_NBGL)
-		set_light_confirmation(true);
-		display_warning(
-		        "Unusual request",
-		        this_fn,
-		        respond_with_user_reject
-		);
-		#endif // HAVE_BAGL
+		_displayUnusualRequestWarning(this_fn);
 	}
 	UI_STEP(DISPLAY_UI_STEP_INSTRUCTIONS) {
-		#ifdef HAVE_BAGL
-		ui_displayPaginatedText(
-		        "Verify address",
-		        "Make sure it agrees with your computer",
-		        this_fn
-		);
-		#elif defined(HAVE_NBGL)
-		set_light_confirmation(true);
-		display_warning(
-		        "Make sure address matches\nwith your computer",
-		        this_fn,
-		        respond_with_user_reject
-		);
-		#endif // HAVE_BAGL
+		_displayVerifyAddressMsg(this_fn);
 	}
 	UI_STEP(DISPLAY_UI_STEP_SPENDING_INFO) {
 		if (determineSpendingChoice(ctx->addressParams.type) == SPENDING_NONE) {
 			// reward address
 			UI_STEP_JUMP(DISPLAY_UI_STEP_STAKING_INFO);
 		}
-		#ifdef HAVE_BAGL
-		ui_displaySpendingInfoScreen(&ctx->addressParams, this_fn);
-		#elif defined(HAVE_NBGL)
-#define SPENDING_INFO_SIZE MAX(11 + BECH32_PREFIX_LENGTH_MAX + 2 * BECH32_BUFFER_SIZE_MAX, 2 * BECH32_BUFFER_SIZE_MAX)
-		char line1[30] = {0};
-		char spendingInfo[SPENDING_INFO_SIZE] = {0};
-		ui_getSpendingInfoScreen(line1, SIZEOF(line1), spendingInfo, SIZEOF(spendingInfo), &ctx->addressParams);
-		fill_address_data(line1, spendingInfo);
-		UI_STEP_JUMP(DISPLAY_UI_STEP_STAKING_INFO);
```

Comment:
I removed this because it did not make sense from the ui machine (we are not jumping anywhere, just proceeding to the next step).

But without it, `this_fn` is not used within the function `_displaySpendingInfo_displayAddr` I replaced it with, so it won't work. Is there a mechanism for scheduling `this_fn` as a callback? (That's what happens e.g. within `ui_displaySpendingInfoScreen`.)

Reply:
I fixed it by adding a callback parameter in `fill_address_data` function